torchtitan/checkpoint.py

```
@@ -435,10 +434,15 @@ def load(self, step: int = -1) -> bool:
         }
         logger.info(f"Loading the checkpoint at step {step}.")
         begin = time.monotonic()
+        shadow_states = {k: v for k, v in states.items() if k not in self.exclude}
+        for exclude_key in self.exclude:
+            if exclude_key not in states:
+                logger.warning(f"{exclude_key} not found in state_dict, skipping")
```

We should just raise an exception. So a better way to do this is
```
if not set(self.exclude).issubset(set(states.keys()):
     raise ValueError("...")
```